tests/foreman/installer/test_installer.py

```diff
@@ -1359,11 +1378,6 @@ def test_positive_check_installer_services(target_sat):
 
     :CaseLevel: System
     """
-    for service in SATELLITE_SERVICES:
-        result = target_sat.execute(f'systemctl status {service}')
-        assert result.status == 0
-        assert 'Active: active (running)' in result.stdout
-
     # check status reported by hammer ping command
     result = target_sat.execute('hammer ping')
```

Comment:
I was wondering: isn't there JSON output that could be used instead of parsing the human readable output? Or is it done intentionally?

Reply:
That's a great question. Ever since I've been around it's always been the human readable output. @JacobCallahan do you know if anyone uses JSON output?

Reply:
if a command can do json output, people are free to use that. however, you will need to strip any extra content and run it through `json.loads`. Unless you're validating the structure of the output, it is almost always much more efficient to assert on the presence of what you're actually looking for.

Reply:
It kind of makes sense, but in this particular case it looks like it's parsing the whole command output and builds a structure. So here I think it makes more sense to ask for the JSON output (if there is any, I didn't check) and use it. The parsing code here could easily break.

Reply:
If you're willing to take that on, then please do! ❤️

Reply:
I took a quick look, but I don't really have time for it. Still, it's good to know the reasoning and goals.

In a similar sense: has there been any thought on using [apypie](https://github.com/Apipie/apypie) as an API client instead of Nailgun? Or perhaps as a backend to Nailgun. It's driving foreman-ansible-modules, so you'd use an officially supported client.

Reply:
@ekohl no worries on doing that now, but may be something to consider more in the future.

I've heard of apypie a few times in the past, but haven't seen examples of it actually in use, so I can't say whether or not it is a potential replacement for nailgun.

Reply:
@ekohl I would be more than happy to see an example how we could incorporate apypie into our framework.

Reply:
@evgeni perhaps you could share your thoughts on this.